qtconsole/tests/test_ansi_code_processor.py

```diff
@@ -49,6 +49,9 @@ def test_colors(self):
             elif i == 2:
                 self.assertEqual(substring, 'last')
                 self.assertEqual(self.processor.foreground_color, None)
+            elif i == 3:
+                self.assertEqual(substring, 'Yellow')
+                self.assertEqual(self.processor.foreground_color, 3)
             else:
                 self.fail('Too many substrings.')
         self.assertEqual(i, 2, 'Too few substrings.')
```

Comment:
Missing an update over the number of substrings here:
```suggestion
        self.assertEqual(i, 3, 'Too few substrings.')
```